drivers/wireless/ieee80211/bcm43xxx/bcmf_driver.c

```diff
@@ -666,300 +669,247 @@ void bcmf_wl_scan_event_handler(FAR struct bcmf_dev_s *priv,
+    {
+      iwr->u.data.length = len;
+      return -E2BIG;
+    }
 
-          if (ie_frame_size > bss->ie_length - ie_offset)
-            {
-              /* Entry too big */
+  start = iwr->u.data.pointer;
+  stop = (FAR char *)iwr->u.data.pointer + iwr->u.data.length;
 
-              break;
-            }
+  /* Sort list by RSSI */
 
-          switch (ie_buffer[ie_offset])
+  for (i = 0; i < priv->scan_result_entries; i++)
+    {
+      for (j = 0; j + 1 < priv->scan_result_entries - i; j++)
+        {
+          if (scan_result[j]->RSSI < scan_result[j + 1]->RSSI)
             {
-              case IEEE80211_ELEMID_RSN:
-                {
-                  size_t ie_frame_size_aligned;
-                  ie_frame_size_aligned = (ie_frame_size + 3) & -4;
-
-                  wlinfo("found RSN\n");
-
-                  if (result_size < BCMF_IW_EVENT_SIZE(data) +
-                                    ie_frame_size_aligned)
-                    {
-                      break;
-                    }
-
-                  iwe = (struct iw_event *)
-                    &priv->scan_result[priv->scan_result_size];
-
-                  iwe->len = BCMF_IW_EVENT_SIZE(data)+ie_frame_size_aligned;
-                  iwe->cmd = IWEVGENIE;
-                  iwe->u.data.flags = 0;
-                  iwe->u.data.length = ie_frame_size;
-                  iwe->u.data.pointer = (FAR void *)sizeof(iwe->u.data);
-                  memcpy(&iwe->u.data + 1, &ie_buffer[ie_offset],
-                         ie_frame_size);
-
-                  priv->scan_result_size += BCMF_IW_EVENT_SIZE(essid) +
-                                            ie_frame_size_aligned;
-                  result_size -= BCMF_IW_EVENT_SIZE(essid) +
-                                 ie_frame_size_aligned;
-                  break;
-                }
-
-              default:
-                break;
+              info = scan_result[j];
+              scan_result[j] = scan_result[j + 1];
+              scan_result[j + 1] = info;
             }
-
-          ie_offset += ie_buffer[ie_offset + 1] + 2;
         }
+    }
 
-      goto process_next_bss;
-
-    scan_result_full:
-
-      /* Continue instead of break to log dropped AP results */
-
-      wlerr("No more space in scan_result buffer\n");
-
-    process_next_bss:
+  /* Copy scan result */
 
-      /* Process next bss_info */
+  for (i = 0; i < priv->scan_result_entries; i++)
+    {
+      info = scan_result[i];
 
-      len -= bss_info_len;
-      bss = (struct wl_bss_info *)((uint8_t *)bss + bss_info_len);
-      bss_count += 1;
-    }
+      /* Copy BSSID */
 
-wl_escan_result_processed:
+      iwe = (FAR struct iw_event *)start;
+      iwe->cmd = SIOCGIWAP;
+      iwe->u.ap_addr.sa_family = ARPHRD_ETHER;
+      memcpy(&iwe->u.ap_addr.sa_data,
+             info->BSSID.ether_addr_octet, IFHWADDRLEN);
+      start = bcmf_wl_iwe_add_event(start, stop, iwe, IW_EV_LEN(ap_addr));
```

Comment:
let's use iwe directly and remove start/stop

Reply:
Done